include/jwt-cpp/jwt.h

```diff
@@ -2699,130 +2707,133 @@ namespace jwt {
 		 * \throw std::runtime_error If claim was not present
 		 * \throw std::bad_cast Claim was present but not a string (Should not happen in a valid token)
 		 */
-		typename json_traits::string_type get_algorithm() const { return get_jwks_claim("alg").as_string(); }
+		typename json_traits::string_type get_algorithm() const { return get_jwk_claim("alg").as_string(); }
 
 		/**
 		 * Get key id claim
 		 * \return key id as string
 		 * \throw std::runtime_error If claim was not present
 		 * \throw std::bad_cast Claim was present but not a string (Should not happen in a valid token)
 		 */
-		typename json_traits::string_type get_key_id() const { return get_jwks_claim("kid").as_string(); }
+		typename json_traits::string_type get_key_id() const { return get_jwk_claim("kid").as_string(); }
 
 		/**
 		 * Get x5c claim
 		 * \return x5c as string
 		 * \throw std::runtime_error If claim was not present
 		 * \throw std::bad_cast Claim was present but not a string (Should not happen in a valid token)
 		 */
-		typename json_traits::string_type get_x5c() const { return json_traits::as_string(get_jwks_claim("x5c").as_array().front()); } //do not use serialize() instead
+		typename json_traits::string_type get_x5c() const { return json_traits::as_string(get_jwk_claim("x5c").as_array().front()); } //do not use serialize() instead
```

Comment:
I think the consumer will/may need the array in larger systems as per [the rfc 4.7](https://tools.ietf.org/html/rfc7517#section-4.7)

> The PKIX certificate containing the key value MUST be the first
   certificate.  This MAY be followed by additional certificates, with
   each subsequent certificate being the one used to certify the
   previous one.

What do you think about having two functions?

```cpp
typename json_traits::array_type get_x5c() const;
typename json_traits::string_type get_x5c_key_value() const;
```

Secondly, `front` might be a little dangerous! [Calling front on an empty container is undefined](https://en.cppreference.com/w/cpp/container/vector/front)

please throw if empty 💥